public/app/features/manage-dashboards/components/RestoreModal.tsx

```diff
@@ -14,7 +15,9 @@ interface Props {
 }
 
 export const RestoreModal = ({ onConfirm, onDismiss, selectedItems, isLoading, ...props }: Props) => {
-  const numberOfDashboards = selectedItems ? Object.keys(selectedItems.dashboard).length : 0;
+  const { data } = useGetAffectedItemsQuery(selectedItems);
+  const numberOfDashboards = data ? data.dashboard : 0; // TODO: Error handling?
```

Comment:
This would work, but is unnecessary. There's a much simpler solution that we're actually already doing later down in this file.

I believe the problem is due to `selectedItems.dashboard` keeping keys with a false value for dashboards once they've been unselected. We need to filter out false values from the object.